ethers-providers/src/transports/quorum.rs

```diff
@@ -759,73 +815,208 @@ mod tests {
+            let quorum = QuorumProvider::builder()
+                .add_providers(providers)
+                .quorum(Quorum::ProviderCount(1))
+                .build();
+            assert_eq!(
+                quorum.get_quorum_number::<U64>("foo", WrappedParams::Zst).await.unwrap().as_u64(),
+                102
+            );
+        }
+        {
+            let mocks = build_n_mock_providers(5);
 
-        let quorum = QuorumProvider::builder()
-            .add_providers(providers.clone())
-            .quorum(Quorum::ProviderCount(2))
-            .build();
-        assert_eq!(
-            quorum.get_quorum_number::<U64>("foo", WrappedParams::Zst).await.unwrap().as_u64(),
-            101
-        );
+            let responses = [100, 100, 100, 68, 68];
+            for (i, value) in responses.iter().enumerate() {
+                mocks[i].push(Ok(serde_json::to_value(U64::from(*value)).unwrap())).await;
+            }
+            let providers: Vec<_> =
+                mocks.into_iter().map(|mock| WeightedProvider::new(mock)).collect();
+
+            let quorum =
+                QuorumProvider::builder().add_providers(providers).quorum(Quorum::Majority).build();
+            assert_eq!(
+                quorum.get_quorum_number::<U64>("foo", WrappedParams::Zst).await.unwrap().as_u64(),
+                100
+            );
+        }
+    }
 
-        let quorum = QuorumProvider::builder()
-            .add_providers(providers.clone())
-            .quorum(Quorum::ProviderCount(1))
-            .build();
-        assert_eq!(
-            quorum.get_quorum_number::<U64>("foo", WrappedParams::Zst).await.unwrap().as_u64(),
-            102
-        );
+    #[tokio::test]
+    async fn test_get_quorum_number_with_slow_responses_within_threshold() {
+        let mut providers = Vec::new();
+        let test_data = [
+            (500, 200),
+            (500, 100),
+            (500, 100),
+            (800, 200),
+            (800, 200),
+        ];
+        for (millis, value) in test_data {
+            let mock = TestMockProvider::new(Duration::from_millis(millis));
+            mock.push(Ok(serde_json::to_value(U64::from(value)).unwrap())).await;
+            providers.push(WeightedProvider::new(mock));
+        }
 
-        let quorum = QuorumProvider::builder()
-            .add_providers(providers.clone())
-            .quorum(Quorum::Majority)
-            .build();
+        let quorum =
+            QuorumProvider::builder().add_providers(providers).quorum(Quorum::Majority).build();
         assert_eq!(
             quorum.get_quorum_number::<U64>("foo", WrappedParams::Zst).await.unwrap().as_u64(),
-            100
+            200
         );
     }
 
+    #[tokio::test]
+    async fn test_get_quorum_number_with_slow_responses_exceeds_threshold() {
+        let mut providers = Vec::new();
+        let test_data = [
+            (500, 200),
+            (500, 200),
+            (500, 200),
```

Comment:
wonder if we should put like 50 as the block number, and assert that the quorum number should be 50?
as atm if all 5 block numbers ended up being considered (which we don't expect given the behavior of this test), we'd get the quorum number of 200 and tests would still pass